toolkit/tools/pkg/imagecustomizerlib/distrohandler.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"context"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+// PackageType represents the type of package manager
+type PackageType string
+
+const (
+	packageTypeRPM PackageType = "rpm"
+	packageTypeDeb PackageType = "deb"
+)
+
+// DistroName represents the distribution name
+type DistroName string
+
+const (
+	distroNameAzureLinux DistroName = "azurelinux"
+	distroNameFedora     DistroName = "fedora"
+)
+
+// distroHandler represents the interface for distribution-specific configuration
+type distroHandler interface {
+	// Package manager configuration
+	getPackageManagerBinary() string
```

Comment:
Everything except for `managePackages` is an implementation detail of package installation with tdnf/dnf. If you want an interface to handle tdnf/dnf differences, then create a new interface. Don't pollute the distro one.